classes/class-wc-connect-compatibility.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * A class for working around the quircks and different versions of WordPress/WooCommerce
+ */
+
+// No direct access please
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Connect_Compatibility' ) ) {
+
+	class WC_Connect_Compatibility {
+		private static $singleton;
+
+		public static function instance() {
+			if( self::$singleton === null ) {
+				self::$singleton = new WC_Connect_Compatibility();
+			}
+
+			return self::$singleton;
+		}
+
+		public function __construct() {
+		}
+
+		public function get_order_id( WC_Order $order ) {
```

Comment:
Actually, the more I think about this the more I think splitting this up into an interface and two classes would be better.

Performance-wise it would mean a single if-statement instead of one per function call, and it would automatically clarify what you just described without a comment.